clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3149,6 +3147,62 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc, SourceRange CallLoc,
     KernelFunc->setInvalidDecl();
 }
 
+// For a wrapped parallel_for, copy attributes from original
+// kernel to wrapped kernel.
+void Sema::copySYCLKernelAttrs(const CXXRecordDecl *KernelObj) {
+  // Get the operator() function of the wrapper
+  CXXMethodDecl *OpParens = nullptr;
+  for (auto *MD : KernelObj->methods()) {
+    if (MD->getOverloadedOperator() == OO_Call) {
+      OpParens = MD;
+      break;
+    }
+  }
+  assert(OpParens && "invalid kernel object");
+
+  typedef std::pair<FunctionDecl *, FunctionDecl *> ChildParentPair;
```

Comment:
Is the purpose of the call graph to just find the original kernel? Or am I misunderstanding?

Reply:
The original kernel is called from new kernel generated as a wrapper as explained near line 522. We need to search within the wrapper kernel to find the original kernel call, so that its attributes can be copied to the wrapper.